test/balancer_test.go

```diff
@@ -683,87 +682,6 @@ func (s) TestServersSwap(t *testing.T) {
 	}
 }
 
-// TestEmptyAddrs verifies client behavior when a working connection is
-// removed.  In pick first and round-robin, both will continue using the old
-// connections.
```

Comment:
Let's make sure there's still a test for this set of circumstances, but that the RPCs begin failing instead of keep using old connections.

Reply:
Also, I thought we confirmed RR was removing the subconns, so RPCs should have been failing? But this test is validating the RPCs would still succeed?

Reply:
> Also, I thought we confirmed RR was removing the subconns, so RPCs should have been failing? But this test is validating the RPCs would still succeed?

Yes, this test was quite broken because it was using the pick_first clientConn to verify round_robin behaviour:
```golang
	// Confirm several new RPCs succeed on round robin.
	for i := 0; i < 10; i++ {
		if _, err := pfclient.UnaryCall(ctx, &testpb.SimpleRequest{}); err != nil {
			t.Fatalf("UnaryCall(_) = _, %v; want _, nil", err)
		}
		time.Sleep(5 * time.Millisecond)
	}
```

Reply:
> Let's make sure there's still a test for this set of circumstances, but that the RPCs begin failing instead of keep using old connections.

Yes, the newly added test verifies that the channel moves to `TransientFailure` when addresses are removed and that a `WaitForReady` RPC blocks until new addresses are added.

Reply:
> Yes, this test was quite broken because it was using the pick_first clientConn to verify round_robin behaviour:

Oof

Reply:
> Yes, the newly added test verifies that the channel moves to TransientFailure when addresses are removed and that a WaitForReady RPC blocks until new addresses are added.

But only with pick first -- is there one for RR too elsewhere?

Reply:
We do have a couple of tests for `round_robin` that I recently cleaned up:
[1](https://github.com/grpc/grpc-go/blob/master/test/roundrobin_test.go#L149)
[2](https://github.com/grpc/grpc-go/blob/master/test/roundrobin_test.go#L175)